gui/default/syncthing/core/syncthingController.js

```diff
@@ -2778,9 +2778,17 @@ angular.module('syncthing.core')
 
             $scope.restoreVersions.tree.filterNodes(function (node) {
                 if (node.folder) return false;
-                if ($scope.restoreVersions.filters.text && node.key.indexOf($scope.restoreVersions.filters.text) < 0) {
-                    return false;
+
+                if ($scope.restoreVersions.filters.text) {
+                    // Use case-insensitive filter and convert backslashes to
+                    // forward slashes to allow using them as path separators.
+                    var filterText = $scope.restoreVersions.filters.text.toLowerCase().replace(/\\/g, '/');
+                    var versionPath = node.key.toLowerCase().replace(/\\/g, '/');
```

Comment:
This earlier comment still applies I think (not very important nit, but still :) ):

> > Yeah, the tree uses forward slashes which I think come from here:
> 
> Then there's no need/point in changing `versionPath`, right?

As path separators are already slashes everywhere, there's no need for `replace` on `versionPath`.

Reply:
I think this is still required, as @acolomb explained in details above in https://github.com/syncthing/syncthing/pull/8995#issuecomment-1650642030, because otherwise there will be no way to match a filename that has an actual backslash in its name (for whatever reason).

Reply:
I don't undrestand how that comment relates to the (internal) paths created from the tree - seems to refer to the user input exclusively. However I might mess things up, at worst it's pointless but even to my (potentially flawed/partial) understanding the `replace` shouldn't hurt.

Reply:
It's not about path separators. On Linux, there is a possibility of having a file named literally `my\file` (with the backslash character in the name, not a path separator). If there is a force replacement of all backslashes to forward slashes in the filter (regardless of the OS) but no replacement in the version path, then it will be impossible to match `my\file` at all.